integration/system/stats_test.go

```diff
@@ -259,53 +283,45 @@ func jsonBlobHasGTE121NetworkStats(blob map[string]interface{}) bool {
 	return true
 }
 
-func (s *DockerAPISuite) TestAPIStatsContainerNotFound(c *testing.T) {
-	testRequires(c, DaemonIsLinux)
-	apiClient, err := client.NewClientWithOpts(client.FromEnv)
-	assert.NilError(c, err)
-	defer apiClient.Close()
+func TestAPIStatsNoStreamGetCpu(t *testing.T) {
+	skip.If(t, RuntimeIsWindowsContainerd(), "FIXME: Broken on Windows + containerd combination")
 
-	expected := "No such container: nonexistent"
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
 
-	_, err = apiClient.ContainerStats(testutil.GetContext(c), "nonexistent", true)
-	assert.ErrorContains(c, err, expected)
-	_, err = apiClient.ContainerStats(testutil.GetContext(c), "nonexistent", false)
-	assert.ErrorContains(c, err, expected)
-}
+	cID := container.Run(ctx, t, apiClient, container.WithImage("busybox"), container.WithCmd("/bin/sh", "-c", "while true;usleep 100; do echo 'Hello'; done"))
 
-func (s *DockerAPISuite) TestAPIStatsNoStreamConnectedContainers(c *testing.T) {
-	testRequires(c, DaemonIsLinux)
+	resp, body, err := request.Get(ctx, fmt.Sprintf("/containers/%s/stats?stream=false", cID))
+	assert.NilError(t, err)
+	assert.Equal(t, resp.StatusCode, http.StatusOK)
+	assert.Equal(t, resp.Header.Get("Content-Type"), "application/json")
+	assert.Equal(t, resp.Header.Get("Content-Type"), "application/json")
 
-	id1 := runSleepingContainer(c)
-	cli.WaitRun(c, id1)
+	var v *types.Stats
+	err = json.NewDecoder(body).Decode(&v)
+	assert.NilError(t, err)
+	body.Close()
 
-	id2 := runSleepingContainer(c, "--net", "container:"+id1)
-	cli.WaitRun(c, id2)
+	cpuPercent := 0.0
 
-	ch := make(chan error, 1)
-	go func() {
-		resp, body, err := request.Get(testutil.GetContext(c), "/containers/"+id2+"/stats?stream=false")
-		defer body.Close()
-		if err != nil {
-			ch <- err
-		}
-		if resp.StatusCode != http.StatusOK {
-			ch <- fmt.Errorf("Invalid StatusCode %v", resp.StatusCode)
-		}
-		if resp.Header.Get("Content-Type") != "application/json" {
-			ch <- fmt.Errorf("Invalid 'Content-Type' %v", resp.Header.Get("Content-Type"))
-		}
-		var v *types.Stats
-		if err := json.NewDecoder(body).Decode(&v); err != nil {
-			ch <- err
-		}
-		ch <- nil
-	}()
+	if testEnv.DaemonInfo.OSType != "windows" {
+		cpuDelta := float64(v.CPUStats.CPUUsage.TotalUsage - v.PreCPUStats.CPUUsage.TotalUsage)
+		systemDelta := float64(v.CPUStats.SystemUsage - v.PreCPUStats.SystemUsage)
+		cpuPercent = (cpuDelta / systemDelta) * float64(len(v.CPUStats.CPUUsage.PercpuUsage)) * 100.0
+	} else {
+		// Max number of 100ns intervals between the previous time read and now
+		possIntervals := uint64(v.Read.Sub(v.PreRead).Nanoseconds()) // Start with number of ns intervals
+		possIntervals /= 100                                         // Convert to number of 100ns intervals
+		possIntervals *= uint64(v.NumProcs)                          // Multiple by the number of processors
```

Comment:
I'm guessing this code was to replicate the logic present in the CLI. That's definitely a part of the code we should make more reusable (part of an SDK or at least some reusable utilities package); 

- https://github.com/docker/cli/blob/623d9b1f68171e3ff9899b21392ac4f102ba533b/cli/command/container/stats_helpers.go#L100-L114
- https://github.com/docker/cli/blob/623d9b1f68171e3ff9899b21392ac4f102ba533b/cli/command/container/stats_helpers.go#L166-L199

Reply:
Indeed! 😅 we should track this somewhere as it relates to the sdk/general refactor, or at least consider this in the big "move logic from the CLI into the client" topic.